habitat/tasks/nav/nav_task.py

```diff
@@ -691,35 +720,166 @@ def update_fog_of_war_mask(self, agent_position):
-            sensor_type = registry.get_sensor(sensor_cfg.TYPE)
-            assert sensor_type is not None, "invalid sensor type {}".format(
-                sensor_cfg.TYPE
-            )
-            task_sensors.append(sensor_type(sim, sensor_cfg))
-
-        self.sensor_suite = SensorSuite(task_sensors)
-        super().__init__(config=task_config, sim=sim, dataset=dataset)
+        super().__init__(config=config, sim=sim, dataset=dataset)
 
     def overwrite_sim_config(
         self, sim_config: Any, episode: Type[Episode]
     ) -> Any:
         return merge_sim_episode_config(sim_config, episode)
+
+    def _check_episode_is_active(self, *args: Any, **kwargs: Any) -> bool:
+        if not hasattr(self, "is_stop_called"):
+            return True
+        else:
+            return not self.is_stop_called
+
+
+class SimulatorAction(Action):
+    def __init__(
+        self,
+        *args: Any,
+        config: Config,
+        sim: Simulator,
+        task: EmbodiedTask,
+        **kwargs: Any
+    ) -> None:
+        self._config = config
+        self._task = task
+        self._sim = sim
+
+    def get_action_space(self):
+        return EmptySpace()
+
+    def reset(self, *args: Any, **kwargs: Any) -> None:
+        return None
+
+
+@registry.register_task_action
+class MoveForwardAction(SimulatorAction):
+    name: str = "MOVE_FORWARD"
+
+    def _get_uuid(self, *args: Any, **kwargs: Any) -> str:
+        return self.name
+
+    def step(self, *args: Any, **kwargs: Any):
+        r"""Update ``_metric``, this method is called from ``Env`` on each
+        ``step``.
+        """
+        return self._sim.step(SimulatorActions.MOVE_FORWARD)
+
+
+@registry.register_task_action
+class TurnLeftAction(SimulatorAction):
+    def _get_uuid(self, *args: Any, **kwargs: Any) -> str:
+        return "TURN_LEFT"
+
+    def step(self, *args: Any, **kwargs: Any):
+        r"""Update ``_metric``, this method is called from ``Env`` on each
+        ``step``.
+        """
+        return self._sim.step(SimulatorActions.TURN_LEFT)
+
+
+@registry.register_task_action
+class TurnRightAction(SimulatorAction):
+    def _get_uuid(self, *args: Any, **kwargs: Any) -> str:
+        return "TURN_RIGHT"
+
+    def step(self, *args: Any, **kwargs: Any):
+        r"""Update ``_metric``, this method is called from ``Env`` on each
+        ``step``.
+        """
+        return self._sim.step(SimulatorActions.TURN_RIGHT)
+
+
+@registry.register_task_action
+class StopAction(SimulatorAction):
+    name: str = "STOP"
+
+    def reset(self, *args: Any, **kwargs: Any):
+        self._task.is_stop_called = False
+
+    def _get_uuid(self, *args: Any, **kwargs: Any) -> str:
+        return self.name
+
+    def step(self, *args: Any, **kwargs: Any):
+        r"""Update ``_metric``, this method is called from ``Env`` on each
+        ``step``.
+        """
+        self._task.is_stop_called = True
+        return self._sim.get_observations_at()
```

Comment:
Yep, looks like the stuff about the stop action can be removed from `simulator.step`